modules/terraform/output.go

```diff
@@ -262,6 +262,32 @@ func OutputForKeys(t testing.TestingT, options *Options, keys []string) map[stri
 	return out
 }
 
+// OutputStruct calls terraform output for the given variable and stores the
+// result in the value pointed to by out. If out is nil or not a pointer, or if
+// the value returned by Terraform is not appropriate for a given target type,
+// it fails the test.
+func OutputStruct(t testing.TestingT, options *Options, key string, out interface{}) {
+	err := OutputStructE(t, options, key, out)
+	require.NoError(t, err)
+}
+
+// OutputStructE calls terraform output for the given variable and stores the
+// result in the value pointed to by v. If v is nil or not a pointer, or if
+// the value returned by Terraform is not appropriate for a given target type,
+// it returns an error.
+func OutputStructE(t testing.TestingT, options *Options, key string, v interface{}) error {
+	out, err := RunTerraformCommandAndGetStdoutE(t, options, "output", "-no-color", "-json", key)
+	if err != nil {
+		return err
+	}
+
+	if err := json.Unmarshal([]byte(out), &v); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
NIT: you can probably simplify this a bit:

```suggestion
	return json.Unmarshal([]byte(out), &v)
```